src/shared/src/backend-supports/project-supports.ts

```diff
@@ -90,11 +91,29 @@ const calculateProjectStartDate = (wps: { duration: number; startDate: Date }[])
   return minDate;
 };
 
+// calculate the project's status based on its pacakges' status
+const calculateProjectStatus = (proj: { workPackages: { wbsElement: { status: WBS_Element_Status } }[] }) => {
+  let isActive = false;
+  let isComplete = true;
+
+  if (proj.workPackages.length === 0) return WbsElementStatus.Inactive;
+
+  proj.workPackages.forEach((wp) => {
+    isComplete = isComplete && wp.wbsElement.status === WbsElementStatus.Complete;
+    isActive = isActive || wp.wbsElement.status === WbsElementStatus.Active;
+  });
+
+  if (isComplete) return WbsElementStatus.Complete;
```

Comment:
a cleaner way to do this logic is using the `.every` function for a list which checks if every elemetn in a list passes a test

so if proj.workPackages.every(wp => wp is complete) then return Complete

then for active you can check that at least one has the active status by checking proj.workPackages.indexOf(wp => status === active) !== -1